src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -7,19 +7,22 @@
 using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
 using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
 using Microsoft.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
 using WellKnownType = Microsoft.AspNetCore.App.Analyzers.Infrastructure.WellKnownTypeData.WellKnownType;
 
 namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel;
 
 internal class EndpointParameter
 {
-    public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
+    public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnownTypes wellKnownTypes)
     {
         Type = parameter.Type;
         Name = parameter.Name;
         Ordinal = parameter.Ordinal;
         Source = EndpointParameterSource.Unknown;
         IsOptional = parameter.IsOptional();
+        IsArray = TryGetArrayElementType(parameter, out var elementType);
```

Comment:
Why do we calculate this ahead of time instead of inline in the if-statement?

Reply:
Because I need the `IsArray` flag in various places within the emitter as well so I need to hold onto the value (the `elementType` is also used in a few places. So even though its only used once or so in this method its used elsewhere multiple times.